pandas/core/arrays/string_arrow.py

```diff
@@ -475,18 +475,14 @@ def _str_count(self, pat: str, flags: int = 0):
         return self._convert_int_dtype(result)
 
     def _str_find(self, sub: str, start: int = 0, end: int | None = None):
-        if start != 0 and end is not None:
-            slices = pc.utf8_slice_codeunits(self._pa_array, start, stop=end)
-            result = pc.find_substring(slices, sub)
-            not_found = pc.equal(result, -1)
-            offset_result = pc.add(result, end - start)
-            result = pc.if_else(not_found, result, offset_result)
-        elif start == 0 and end is None:
-            slices = self._pa_array
-            result = pc.find_substring(slices, sub)
-        else:
+        if (
+            pa_version_under13p0
+            and not (start != 0 and end is not None)
+            and not (start == 0 and end is None)
+        ):
+            # https://github.com/pandas-dev/pandas/pull/59562/files#r1725688888
```

Comment:
This link points to this PR, but it doesn't seem to work to actually link to a specific comment

Reply:
Ah, I suppose it points to https://github.com/pandas-dev/pandas/pull/59562#discussion_r1725688888

Now, can't we move that into the mixin as well? (avoiding this override) This is something that was just buggy in pyarrow before that version AFAIU, so I think there is no harm in also doing object-dtype fallback for ArrowDtype, since it otherwise just errors wrongly

Reply:
huh i couldve sworn i copy/pasted it from somewhere, but now i cant find it.  will update

Reply:
Can you check my question in my second comment above?

Reply:
About moving the implementation to the mixin?  This PR now does that.

Reply:
It's about the `if` block that this thread is commenting on, that is not in the mixin and my question is whether it shouldn't be moved as well? (it's a bug that affects ArrowEA as well AFAIU)

Reply:
thanks for clarifying.  will give it a try.